CHANGELOG.md

```diff
@@ -1,3 +1,124 @@
+
+The `@commercetools-frontend/application-shell-connectors` used to contain some connectors that have now been **removed**:
+
+- **configuration**: `injectConfiguration` and `<ConfigurationConsumer>`
+- **project-data-locale**: `withProjectDataLocale` and `<GetProjectDataLocale>`
+- **user-permissions**: `withUserPermissions` and `<GetUserPermissions>`
+- **user-time-zone**: `withUserTimeZone` and `<GetUserTimeZone>`
+
+As a replacement, we now have a single component `<GetApplicationState>` and `withApplicationState` that provide all the necessary "global" information about the `user`, `project` and application `environment`.
+
+#### `user` fields
+
+- `id`
+- `email`
+- `firstName`
+- `lastName`
+- `locale`
+- `timeZone`
+
+#### `project` fields
+
+- `key`
+- `version`
+- `name`
+- `countries`
+- `currencies`
+- `languages`
+- `permissions`: an object containing boolean flags about the permissions of the logged in user for the selected project (e.g. `{ canViewProducts: true, canManageOrders: false, ... }`)
+- `dataLocale`: the selected project **locale** (from the locale switcher in the AppBar) used to render a localized field of the project data. The available values are based on the `project.languages`
+
+#### `environment` fields
+
+This object contains application specific environment information defined in the `env.json`. The object will then be available on runtime from `window.app`. However, to avoid accessing those values globally, we inject this object into the application context.
+
+The following are common fields defined in `env.json`. However, each application can provide more specific fields that cannot be documented.
+
+- `frontendHost`: the host where the Merchant Center application is running (e.g. `mc.commercetools.com`)
+- `mcApiUrl`: the API URL of the Merchant Center (`https://mc-api.commercetools.com` for projects in `EU` and `https://mc-api.commercetools.co` for projects in `US`)
+- `location`: the location where the Merchant Center is running, usually `eu` or `us`
+- `env`: the environment where the Merchant Center is running, usually `production` or `staging`
+- `cdnUrl`: the URL where the static assets are stored
+- `servedByProxy`: a flag to indicate if this application is running behind the Merchant Center proxy or not, usually `true` for production and `false` for local development
+
+#### Usage example
+
+```js
+<GetApplicationState
+  render={({ user, project, environment }) => (
+    <div>{...}</div>
+  )}
+/>
+```
+
+You can also use the HOC `withApplicationState` that will inject a `applicationState` prop.
+
+```js
+withApplicationState()(MyComponent);
+```
+
+...or pass a mapping function as the first argument to return custom shape of the injected props
+
+```js
+withApplicationState(applicationState => ({
+  projectKey: applicationState.project && applicationState.project.key,
+  userEmail: applicationState.user && applicationState.user.email,
+}))(MyComponent);
+```
```

Comment:
This documentation is nice! I'd move it to a more permanent place like a dedicated `.md` file and link to it from the  `README`. Then I would remove some details form the change log and also link to them from here.

I'm currently writing some documentation about `test-utils` myself. Separate docs for `ApplicationState` would be nice. I could also link to them from `test-utils` then 👍

Reply:
Feel free to do those changes 😉

Reply:
;)

Reply:
...but he already have this documentation here or what part are we missing? https://github.com/commercetools/merchant-center-application-kit/tree/master/packages/application-shell-connectors/src/components/application-state

Reply:
Nice, this is exactly what I was looking for! I'd like to it from the change log though to make easier to discover.

I finished writing testing-docs now.  They are available at #64. I would link to them from the change log as well, as we improved our test-utils too.